p2p/discover/v4_udp.go

```diff
@@ -719,6 +727,14 @@ func (t *UDPv4) handleFindnode(h *packetHandlerV4, from *net.UDPAddr, fromID eno
 	closest := t.tab.closest(target, bucketSize, true).entries
 	t.tab.mutex.Unlock()
 
+	if len(closest) == 0 {
+		t.tab.mutex.Lock()
+		closest = t.tab.closest(target, bucketSize, false).entries
+		t.tab.mutex.Unlock()
```

Comment:
Wouldn't it be simpler to do this check before releasing the mutex, instead of mutex-release, check, mutex-grab, release ?

Reply:
Well, it depends: I assumed that this is an edge-case that happens almost never on "normal" networks (but is required to allow small networks to work). So I didn't want to hold the mutex longer any longer than needed in normal, non-edge-cases. 

But if we don't mind the additional `len(closest)== 0` in the mutex I'm ok with doing everything in a single mutex-grab.